materialize-starburst-galaxy/galaxy.go

```diff
@@ -0,0 +1,430 @@
+package main
+
+import (
+	"context"
+	stdsql "database/sql"
+	"encoding/json"
+	"fmt"
+	m "github.com/estuary/connectors/go/protocols/materialize"
+	boilerplate "github.com/estuary/connectors/materialize-boilerplate"
+	sql "github.com/estuary/connectors/materialize-sql"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	pm "github.com/estuary/flow/go/protocols/materialize"
+	log "github.com/sirupsen/logrus"
+	_ "github.com/trinodb/trino-go-client/trino"
+	"go.gazette.dev/core/consumer/protocol"
+	"net/url"
+	"time"
+)
+
+// config represents the endpoint configuration for galaxy.
+// It must match the one defined for the source specs (flow.yaml) in Rust.
+type config struct {
+	Host               string `json:"host" jsonschema:"title=Host and optional port" jsonschema_extras:"order=0"`
+	Catalog            string `json:"catalog" jsonschema:"title=Catalog" jsonschema_extras:"order=1"`
+	Schema             string `json:"schema" jsonschema:"title=Schema" jsonschema_extras:"order=2"`
+	Account            string `json:"account" jsonschema:"title=Account" jsonschema_extras:"order=3"`
+	Password           string `json:"password" jsonschema:"title=Password" jsonschema_extras:"secret=true,order=4"`
+	AWSAccessKeyID     string `json:"awsAccessKeyId" jsonschema:"title=Access Key ID" jsonschema_extras:"order=5"`
+	AWSSecretAccessKey string `json:"awsSecretAccessKey" jsonschema:"title=Secret Access Key" jsonschema_extras:"secret=true,order=6"`
+	Region             string `json:"region" jsonschema:"title=Region" jsonschema_extras:"order=7"`
+	Bucket             string `json:"bucket" jsonschema:"title=Bucket" jsonschema_extras:"order=8"`
+	DataStoreDir       string `json:"dataStoreDir" jsonschema:"title=Directory for storing temporary files" jsonschema_extras:"order=8"`
+
+	Advanced advancedConfig `json:"advanced,omitempty" jsonschema:"title=Advanced Options,description=Options for advanced users. You should not typically need to modify these." jsonschema_extras:"advanced=true"`
+}
+
+type advancedConfig struct {
+	UpdateDelay string `json:"updateDelay,omitempty" jsonschema:"title=Update Delay,description=Potentially reduce active warehouse time by increasing the delay between updates. Defaults to 30 minutes if unset.,enum=0s,enum=15m,enum=30m,enum=1h,enum=2h,enum=4h"`
+}
+
+// ToURI converts the Config to a DSN string.
+func (c *config) ToURI() string {
+
+	var params = make(url.Values)
+	params.Add("catalog", c.Catalog)
+	params.Add("schema", c.Schema)
+
+	var uri = url.URL{
+		Scheme:   "https",
+		Host:     c.Host,
+		User:     url.UserPassword(c.Account, c.Password),
```

Comment:
Can there be avoided the exposure of the user and password in the url? Headers maybe?